pkg/oauth/apis/oauth/validation/validation_test.go

```diff
@@ -746,3 +746,38 @@ func TestValidateAuthorizeTokensUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateUserNameField(t *testing.T) {
+	fieldArg := field.NewPath("metadata", "name")
+	valid := "kube:admin"
+	errs := ValidateUserNameField(valid, fieldArg)
+	if len(errs) != 0 {
+		t.Errorf("expected success: %v", errs)
+	}
+
+	errorCases := map[string]struct {
+		name string
+		T    field.ErrorType
+		F    string
```

Comment:
don't hesitate to make this more verbose, i.e. `expected_err_type` and `expected_err_field`.

Reply:
Done